src/V1/APIs/Services/Auth/auth.service.ts

```diff
@@ -0,0 +1,220 @@
+const mail = new Email();
+
+dotenv.config({ path: './src/V1/APIs/Config/.env' });
+
+export default class AuthService {
+    public async registerUser(req: Request, res: Response, next: NextFunction) {
+        try {
+            const { firstName, lastName, password, phoneNumber, email } =
+                req.body;
+            const code = crypto.randomInt(100000, 1000000);
+            const user = {
+                first_name: firstName,
+                last_name: lastName,
+                password_digest: password,
+                phone_number: phoneNumber,
+                email,
+                verification_code: code,
+            };
+            const userEmail = await validation.validateEmail(user.email);
+            if (userEmail) {
+                return res.status(400).json({
+                    success: false,
+                    error:' Email is already taken',
+                    message: 'Registration failed!',
+                });
+            }
+            const userPhone = await validation.validatePhoneNumber(
+                user.phone_number
+            );
+            if (userPhone) {
+                return res.status(400).json({
+                    success: false,
+                    error:'Phone number is already taken',
+                    message: 'Registration failed!',
+                });
+            }
+            if (userEmail === undefined && userPhone === undefined) {
+                const registerUser: UserType = await authRepository.createUser(
+                    user
+                );
+                if (!registerUser) {
+                    return res.status(400).json({
+                        success: false,
+                        error:'Unable to create user',
+                        message: 'Registration failed',
+                    });
+                }
+                if (registerUser) {
+                    const message = `<p>Hello ${registerUser.first_name},</p>
+                <p>Welcome to DeliveryCog Platform. Please verify your 
+                email address with the OTP code below. It would expire after 10mins.<p>
+                <p>OTP: <b>${registerUser.verification_code}</b></p>`;
+                    const userInfo = {
+                        first_name: registerUser.first_name,
+                        email: registerUser.email,
+                        subject: 'Verify your DeliveryCog Account',
+                        code: Number(registerUser.verification_code),
+                        message,
+                    };
+
+                    await mail.sendOTP(userInfo);
+                    return res.status(201).json({
+                        success: true,
+                        message:
+                            'Account successfully created, Check your mail for activation code',
+                    });
+                }
+            }
+        } catch (error) {
+            console.log(error);
```

Comment:
Replace this by using pretty logger module

Reply:
Not supposed to be there I forgot to remove it after checking for an error. thanks.